pkg/virt-controller/watch/vmi.go

```diff
@@ -729,6 +745,17 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		if !vmi.IsUnprocessed() {
 			return nil
 		}
+		// let's check if we already have topology hints or if we are still waiting for them
+		if vmi.Status.TopologyHints == nil {
```

Comment:
could we make a function in topologyHinter called `topologyHinter.RequiresTopologyHints(vmi)` which just tells us if topology hints are required for the VMI or not.

Then this if check could be

```
if vmi.Status.TopologyHints == nil && c.topologyHinter.TopologyHintsRequiredForVMI(vmi) {
    log.Log.V(3).Object(vmi).Infof("Delaying pod creation until topology hints are set")
    // no hints in the status yet, but apparently the VMI requires some, let's retry when they are there
    return nil
}
```

One reason I suggest that is that this logic is competing with UpdateStatus a bit. In the event that the topology hints aren't found, I think we're firing this ```FailedGatherhingClusterTopologyHints``` event twice for each reconcile loop.

Reply:
:+1: Fixed.